docs/eager-inference-annotations-for-polymorphic-types.md

```diff
@@ -0,0 +1,72 @@
+# Eager Inference Annotations for Polymorphic Types
+
+## Summary  
+
+The RFC introduces a feature to annotate polymorphic function types to express that the first bind to `T` will be the one that sticks. 
```

Comment:
What would be the behavior when a type references `T` more than once, such as an object?

```luau
function F<T!>(value: { a: T, b: T })
```

Additionally, what about when the object type is aliased (i.e there's no specific "first occurence")

```luau
type MyObject<T> = { a: T, b: T };

function F<T!>(value: MyObject<T>)
```

Reply:
> What would be the behavior when a type references `T` more than once, such as an object?

It would behave in the same way as the old type solver which is taking the first definition that binds to `T`. 

> Additionally, what about when the object type is aliased (i.e there's no specific "first occurence")

Similar answer to the last point which is mirroring the old behaviour but in this case would be to go in the order of the fields. I don't know the exact heuristics for that is but I feel most of this could be answered with just saying it should do whatever the type solver did. 

I will note that I do think something like `noinfer<T>` is a pretty good solution in these cases but I haven't personally prepared to make a case for that proposal.